### PR TITLE
chore: migrate to main branch

### LINKED ISSUE #247: Migrate to main branch
Contributing to googleapis/google-cloud-python#10579, switching the default branch to main branch. Also replacing all occurrences of master to main.

### PR DESCRIPTION
**Pull request by dandhlee**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-dataproc/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #247  🦕

### CODE REVIEW

File: samples/AUTHORING_GUIDE.md [hunk 1]
```diff
@@ -1 +1 @@
-See https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md
\ No newline at end of file
+See https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/AUTHORING_GUIDE.md
```
Comment (parthea) [inline samples/AUTHORING_GUIDE.md:1]: I get 404 with the new link

File: UPGRADING.md [hunk 2]
```diff
@@ -69,7 +69,7 @@ In `google-cloud-dataproc<2.0.0`, parameters required by the API were positional
 
 In the 2.0.0 release, all methods have a single positional parameter `request`. Method docstrings indicate whether a parameter is required or optional.
 
-Some methods have additional keyword only parameters. The available parameters depend on the [`google.api.method_signature` annotation](https://github.com/googleapis/googleapis/blob/master/google/cloud/dataproc/v1/clusters.proto#L88) specified by the API producer.
+Some methods have additional keyword only parameters. The available parameters depend on the [`google.api.method_signature` annotation](https://github.com/googleapis/googleapis/blob/main/google/cloud/dataproc/v1/clusters.proto#L88) specified by the API producer.
```
Comment (parthea) [inline UPGRADING.md:72]: I get 404 with the new link

File: README.rst [hunk 3]
```diff
@@ -9,7 +9,7 @@ Python Client for Google Cloud Dataproc API
 - `Product Documentation`_
 
 .. |ga| image:: https://img.shields.io/badge/support-ga-gold.svg
-   :target: https://github.com/googleapis/google-cloud-python/blob/master/README.rst#ga-support
+   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#ga-support
```
Comment (parthea) [inline README.rst:12]: I get 404 with the new link

**Review by parthea — COMMENTED** [review]
Minor observations, otherwise LGTM.

File: samples/AUTHORING_GUIDE.md (hunk 1 above)
Comment (dandhlee) [reply samples/AUTHORING_GUIDE.md]: Done.

File: UPGRADING.md (hunk 2 above)
Comment (dandhlee) [reply UPGRADING.md]: Done.

File: README.rst (hunk 3 above)
Comment (dandhlee) [reply README.rst]: Done.

**Comment by dandhlee** [comment]
Please take a look again!

**Review by parthea — APPROVED** [review]

**Pull request merged**